src/common/mfu_io.c

```diff
@@ -2315,7 +2313,8 @@ ssize_t daos_llistxattr(const char* path, char* list, size_t size, mfu_file_t* m
     else {
         /* lookup and open name */
         dfs_obj_t* obj;
-        int rc = dfs_lookup_rel(mfu_file->dfs, parent, name, O_RDWR, &obj, NULL, NULL);
+        int lookup_flags = O_RDWR | O_NOFOLLOW;
+        int rc = dfs_lookup_rel(mfu_file->dfs, parent, name, lookup_flags, &obj, NULL, NULL);
```

Comment:
This might be the spot I noticed before, where it looks like dfs_release might be called after an error.

Reply:
The release for this checks `if (rc && (got_size != -1))`, so release isn't called after an error from dfs_lookup_rel.
But... release sometimes isn't called after an error from dfs_listxattr, but it SHOULD be, because that handle is still open.
Thanks for noticing this!

Reply:
Edit: No, you're right. It WAS being called after dfs_lookup failed 👍